### ISSUE TITLE
Nx unable to run multiple cypress features simultaneously 

### ISSUE THREAD

**Issue by DanLatimer** [nrwl/nx#22109]
### Current Behavior

Unable to run multiple project's cypress e2e tasks at once, the ports will collide 

### Expected Behavior

Should be able to host a SPA with a single static file serve and run multiple cypress tasks against it

### GitHub Repo

https://github.com/DanLatimer/nx-cypress-multirun

### Steps to Reproduce

1. clone the repo
2. install pnpm (if you don't have it)
3. run pnpm i
4. run `pnpm nx run-many -t e2e`

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 20.9.0
OS     : darwin-arm64
pnpm   : 8.10.5

nx                 : 18.0.6
@nx/js             : 18.0.6
@nx/jest           : 18.0.6
@nx/linter         : 18.0.6
@nx/eslint         : 18.0.6
@nx/workspace      : 18.0.6
@nx/cypress        : 18.0.6
@nx/devkit         : 18.0.6
@nx/eslint-plugin  : 18.0.6
@nx/react          : 18.0.6
@nrwl/tao          : 18.0.6
@nx/vite           : 18.0.6
@nx/web            : 18.0.6
typescript         : 5.3.3
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by djorgji** [comment]
Same thing happens with e2e-ci inferred tasks.

Beyond each e2e task trying to start the static server on its own, if you have `setupNodeEvents` as document [here](https://nx.dev/recipes/cypress/cypress-setup-node-events) it times out (15s global timeout) if the static server takes too long to come online (code needs to run build), and it doesn't seem to use the remote cache properly in nx agents.

``` ts
const nxConfig = nxE2EPreset(__dirname, {
  cypressDir: 'src',
  bundler: 'webpack',
  webServerCommands: {
    default: 'nx run app:serve',
    production: 'nx run app:serve:production',
  },
  ciWebServerCommand: 'nx run app:serve-static',
});

export default defineConfig({
  e2e: {
    ...nxConfig,
    ...cypressJsonConfig,
    baseUrl: 'http://localhost:4200',
    env: {
      envName: process.env.USER_BRANCH,
    },
    async setupNodeEvents(on, config) {
      await nxConfig.setupNodeEvents(on, config);

      on('task', {
        getGmailMessages: async (args) => {
          // custom task
        },
      });
    },
  },
```

```
222  app-e2e: Your configFile threw an error from: /home/workflows/workspace/apps/app-e2e/cypress.config.ts
223  app-e2e: 
224  app-e2e: The error was thrown while executing your e2e.setupNodeEvents() function:
225  app-e2e: 
226  app-e2e: Error: Web server failed to start in 15000ms. This can be configured in cypress.config.ts.
227  app-e2e:     at Timeout._onTimeout (/home/workflows/workspace/node_modules/@nx/cypress/plugins/cypress-preset.js:124:20)
228  app-e2e:     at listOnTimeout (node:internal/timers:573:17)
229  app-e2e:     at processTimers (node:internal/timers:514:7)
```

**Comment by FrozenPandaz** [comment]
Hm, the way we've solved this with the executor is that the dev-server target could be set to `auto` where it will find an available port.

However, with the cypress preset/ inferred tasks, it is hard coded what the `baseUrl` is... We may need a way to dynamically determine the `baseUrl` after the server is running.

The other thing which may be able to help with this is [sidecar tasks](https://github.com/nrwl/nx/discussions/23273) which I've just created a RFC for. But.. I do think we should be able to solve this issue like we have in the past without sidecar tasks.
